### PR TITLE
Initial draft of logging api

### PR DESCRIPTION
**Pull request by fables-tales**

### CODE REVIEW

File: src/logger/mod.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt::Display;
+use serde_json::Value;
+
+use ::Level;
+use ::backend::Backend;
+
+/// Logger defines the core interface for ceethane loggers
+pub trait Logger: Clone {
```
Comment (sgrif) [inline src/logger/mod.rs]: Clone as a supertrait is considered an antipattern

File: src/backend/stdout.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+use serde_json::{Value};
+use time;
+use serde_json;
+
+use super::syslog_time_format;
+use ::Level;
+use ::backend::Backend;
+
+#[derive(Debug, Copy, Clone)]
+pub struct Stdout {
```
Comment (sgrif) [inline src/backend/stdout.rs:11]: Why not use https://doc.rust-lang.org/nightly/std/io/struct.Stdout.html?

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+#[macro_export]
+macro_rules! logf {
+    (@single $($x:tt)*) => (());
+    (@count $($rest:expr),*) => (<[()]>::len(&[$(logf!(@single $rest)),*]));
+
+    ($($key:expr => $value:expr,)+) => { logf!($($key => $value),+) };
+    ($($key:expr => $value:expr),*) => {
+        {
+            let _cap = logf!(@count $($key),*);
```
Comment (sgrif) [inline src/lib.rs]: You don't need these underscores. Rust's macros are hygienic.

File: src/backend/stdout.rs (hunk 2 above)
Comment (fables-tales) [reply src/backend/stdout.rs:11]: I feel like making an impl for a ceethane backend on `io::Stdout` itself doesn't make anywhere near as much sense as ceethane having it's own struct which is it's stdout backend.

Comment (sgrif) [reply src/backend/stdout.rs:11]: Agree to disagree. :)

**Pull request merged**